### PR TITLE
fix all package exports, remove `/src/` import path segments

### PR DESCRIPTION
**Pull request by turbocrime**
#762 

resolves dev/build issues relevant to #985

this removes all `/src/` import path segments from everything by providing the correct exports config in package.json

many imports throughout the repo have to be modified as a result

### CODE REVIEW

File: packages/polyfills/package.json [hunk 1]
```diff
@@ -9,15 +9,22 @@
...
-      "./src/*": "./dist/*.js",
-      "./src/ReadableStream[Symbol.asyncIterator]": {
+      "./ReadableStream[Symbol.asyncIterator]": {
         "import": "./dist/ReadableStream_Symbol.asyncIterator_.js",
-        "types": "./dist/ReadableStream[Symbol.asyncIterator].d.ts"
+        "types": "./src/ReadableStream_Symbol.asyncIterator_.d.ts"
+      },
+      "./*": {
+        "import": "./dist/*.js",
+        "types": "./dist/*.d.ts"
```
Comment (turbocrime) [inline packages/polyfills/package.json:27]: replacing brackets with underscores for path compatibility concerns

File: packages/polyfills/src/Array.fromAsync.test.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { expect, describe, it } from 'vitest';
 
 import Array from './Array.fromAsync';
-import './ReadableStream[Symbol.asyncIterator]';
+import './ReadableStream_Symbol.asyncIterator_';
```
Comment (turbocrime) [inline packages/polyfills/src/Array.fromAsync.test.ts:4]: internally, underscore paths are literal, everywhere else they remain

File: packages/bech32m/tsconfig.json
Comment (turbocrime) [inline packages/bech32m/tsconfig.json:1]: bech32m build was broken due to failing to exclude dist

File: packages/query/package.json [hunk 3]
```diff
@@ -8,6 +8,10 @@
     "lint": "eslint \"**/*.ts*\"",
     "test": "vitest run"
   },
+  "exports": {
+    "./*": "./src/*.ts",
+    "./quieriers": "./src/queriers/*.ts"
```
Comment (jessepinho) [inline packages/query/package.json:13]:
```suggestion
    "./queriers": "./src/queriers/*.ts"
```

Also, isn't this case handled by `./*` above it?

**Review by jessepinho — APPROVED** [review]
NOICE (one lil typo, then merge)

File: packages/query/package.json (hunk 3 above)
Comment (turbocrime) [reply packages/query/package.json]: it does seem to function without that, so removed

**Pull request merged**